base/sort.jl

```diff
@@ -1285,19 +1285,61 @@ function sort!(A::AbstractArray;
                by=identity,
                rev::Union{Bool,Nothing}=nothing,
                order::Ordering=Forward)
-    ordr = ord(lt, by, rev, order)
+    _sort!(A, Val(dims), alg, ord(lt, by, rev, order))
+end
+function _sort!(A::AbstractArray, ::Val{K}, alg::Algorithm, order::Ordering) where K
     nd = ndims(A)
-    k = dims
 
-    1 <= k <= nd || throw(ArgumentError("dimension out of range"))
+    1 <= K <= nd || throw(ArgumentError("dimension out of range"))
 
-    remdims = ntuple(i -> i == k ? 1 : axes(A, i), nd)
+    remdims = ntuple(i -> i == K ? 1 : axes(A, i), nd)
     for idx in CartesianIndices(remdims)
-        Av = view(A, ntuple(i -> i == k ? Colon() : idx[i], nd)...)
-        sort!(Av, alg, ordr)
+        Av = view(A, ntuple(i -> i == K ? Colon() : idx[i], nd)...)
+        sort!(Av, alg, order)
     end
     A
 end
+"""
+    sortperm(A; dims::Integer, alg::Algorithm=DEFAULT_UNSTABLE, lt=isless, by=identity, rev::Bool=false, order::Ordering=Forward)
+
+Return a permutation vector `I` that puts `v[I]` in sorted order along the given dimension. The order is specified
+using the same keywords as [`sort!`](@ref). The permutation is guaranteed to be stable even
+if the sorting algorithm is unstable, meaning that indices of equal elements appear in
+ascending order.
+
+To sort slices of an array, refer to [`sortslices`](@ref).
+
+# Examples
+```jldoctest
+julia> A = [8 7; 5 6]
+2×2 Matrix{Int64}:
+ 8  7
+ 5  6
+
+julia> sortperm(A, dims = 1)
+2×2 Matrix{Int64}:
+ 2  4
+ 1  3
+
+julia> sortperm(A, dims = 2)
+2×2 Matrix{Int64}:
+ 3  1
+ 2  4
+```
+"""
+function sortperm(A::AbstractArray; dims::Integer,
+                  alg::Algorithm=DEFAULT_UNSTABLE,
+                  lt=isless,
+                  by=identity,
+                  rev::Union{Bool,Nothing}=nothing,
+                  order::Ordering=Forward)
+    ix = Base.copymutable(LinearIndices(A))
+    order = Base.Perm(
+        Base.ord(Base.isless, identity, nothing, Base.Order.Forward),
```

Comment:
```suggestion
       ord(lt, by, rev, order),
```
Use the keyword arguments here